---
title: "Park Flies Microbiome Analysis"
output: learnr::tutorial
runtime: shiny_prerendered
description: "Comparative Analysis of the Microbiome of PD flies and Control flies"
---

```{r setup, include=FALSE}

library(lme4)
library(multcomp)
library(dunn.test)
library(rcompanion)
library(tidyr)
library(ggplot2)
library(gridExtra)
library(reshape)
library(ade4)
library(grid)
library(gtable)
library(PMCMR)
library(multcompView)
library(splitstackshape)
library(readxl)
library(reshape2)
library(survival)
library(maps)
library(dplyr)
library(vegan)
library(cowplot)
library(stringr)
library(exactRankTests)
library(readr)


library(devtools)
library(usethis)
library(roxygen2)
library(learnr)



knitr::opts_chunk$set(echo = FALSE)
```

## Introduction

### Welcome to the Parkinson Flies Microbiome Analysis Figures

In this tutorial you will see how all figures were created for the Parkinson Flies Microbiome Paper. Feel free to experiment with the tutorial by changing the taxon levels or other similar changes.

## Bash Work

### Using QIIME2


source activate qiime2-2019.7

cd ~/Park-Proj/Evan_stuff/parkFlies/Files/CURA_project

name=call19
mapper=call_mapper.tsv
seqsdir=reads/


*filter out wolbachia reads*
qiime taxa filter-table \
  --i-table table-$name"".qza \
  --i-taxonomy taxonomy-$name"".qza \
  --p-exclude Wolbachia \
  --o-filtered-table table-$name""_noW.qza
  
*filter out individual OTUs*
qiime feature-table filter-features\
  --i-table table-$name""_noW.qza \
  --m-metadata-file OTUs.tsv \
  --p-exclude-ids \
  --o-filtered-table table-$name""_feature.qza
  
*filter other unwanted contents*
qiime taxa filter-table \
  --i-table table-$name"".qza \
  --i-taxonomy taxonomy-$name"".qza \
  --p-exclude Mitochondira,Chloroplast,Archaea,Wolbachia \
  --o-filtered-table table-$name""_noOther.qza



bash ../16sscripts/standard_qiime3.sh $name""_noW $name UCLA_controls 164 $mapper "lab = 'UCLA' AND genotype_class = 'control'"

bash ../16sscripts/standard_qiime3.sh $name""_noW $name UCLA_mutants 164 $mapper "lab = 'UCLA' AND genotype_class = 'mutant'"

bash ../16sscripts/standard_qiime3.sh $name""_noW $name UCLA 164 $mapper "lab = 'UCLA'"



bash ../16sscripts/standard_qiime3.sh $name""_noW $name Stanford_controls 2974 $mapper "lab = 'Stanford' AND genotype_class = 'control'"

bash ../16sscripts/standard_qiime3.sh $name""_noW $name Stanford_mutants 2974 $mapper "lab = 'Stanford' AND genotype_class = 'mutant'"

bash ../16sscripts/standard_qiime3.sh $name""_noW $name Stanford 2974 $mapper "lab = 'Stanford'"



bash ../16sscripts/standard_qiime3.sh $name""_noW $name Pallanck_controls 7188 $mapper "lab = 'Pallanck'AND genotype_class = 'control'"

bash ../16sscripts/standard_qiime3.sh $name""_noW $name Pallanck_mutants 7188 $mapper "lab = 'Pallanck'AND genotype_class = 'mutant'"

bash ../16sscripts/standard_qiime3.sh $name""_noW $name Pallanck 7188 $mapper "lab = 'Pallanck'"



bash ../16sscripts/standard_qiime3.sh $name""_noW $name Bellen_controls 1565 $mapper "lab = 'Bellen' AND genotype_class = 'control'"

bash ../16sscripts/standard_qiime3.sh $name""_noW $name Bellen_mutants_174 1565 $mapper "lab = 'Bellen' AND genotype = 'iPLA2-VIA[Delta174]'"

bash ../16sscripts/standard_qiime3.sh $name""_noW $name Bellen_mutants_192 1565 $mapper "lab = 'Bellen' AND genotype = 'iPLA2-VIA[Delta192]'"

bash ../16sscripts/standard_qiime3.sh $name""_noW $name Bellen 1565 $mapper "lab = 'Bellen'"



bash ../16sscripts/standard_qiime3.sh $name""_feature $name All 200 $mapper "lab IN ('UCLA','Stanford','Bellen','Pallanck')"




bash ../16sscripts/standard_qiime3.sh $name""_noW $name iPLA174_noControl 1565 $mapper "genotype IN ('iPLA2-VIA[Delta174]')"

bash ../16sscripts/standard_qiime3.sh $name""_noW $name iPLA174 1565 $mapper "genotype IN ('iPLA2-VIA Control','iPLA2-VIA[Delta174]')"

bash ../16sscripts/standard_qiime3.sh $name""_noW $name Bloom_174 1565 $mapper "lab = 'Bloomington' AND genotype = 'iPLA2-VIA[Delta174]'"



bash ../16sscripts/standard_qiime3.sh $name""_noW $name iPLA192_noControl 1565 $mapper "genotype IN ('iPLA2-VIA[Delta192]')"

bash ../16sscripts/standard_qiime3.sh $name""_noW $name iPLA192 1565 $mapper "genotype IN ('iPLA2-VIA Control','iPLA2-VIA[Delta192]')"

bash ../16sscripts/standard_qiime3.sh $name""_noW $name Bloom_192 1565 $mapper "lab = 'Bloomington' AND genotype = 'iPLA2-VIA[Delta192]'"





















## Topic 1

### Exercise 

*Here's a simple exercise with an empty code chunk provided for entering the answer.*

Write the R code required to add two plus two:

```{r two-plus-two, exercise=TRUE}

```

### Exercise with Code

*Here's an exercise with some prepopulated code as well as `exercise.lines = 5` to provide a bit more initial room to work.*

Now write a function that adds any two numbers and then call it:

```{r add-function, exercise=TRUE, exercise.lines = 5}
add <- function() {
  
}
```

## Topic 2

### Exercise with Hint

*Here's an exercise where the chunk is pre-evaulated via the `exercise.eval` option (so the user can see the default output we'd like them to customize). We also add a "hint" to the correct solution via the chunk immediate below labeled `print-limit-hint`.*

Modify the following code to limit the number of rows printed to 5:

```{r print-limit, exercise=TRUE, exercise.eval=TRUE}
mtcars
```

```{r print-limit-hint}
head(mtcars)
```

### Quiz

*You can include any number of single or multiple choice questions as a quiz. Use the `question` function to define a question and the `quiz` function for grouping multiple questions together.*

Some questions to verify that you understand the purposes of various base and recommended R packages:

```{r quiz}
quiz(
  question("Which package contains functions for installing other R packages?",
    answer("base"),
    answer("tools"),
    answer("utils", correct = TRUE),
    answer("codetools")
  ),
  question("Which of the R packages listed below are used to create plots?",
    answer("lattice", correct = TRUE),
    answer("tools"),
    answer("stats"),
    answer("grid", correct = TRUE)
  )
)
```
