---
title: "Park Flies Microbiome Analysis"
output: learnr::tutorial
runtime: shiny_prerendered
description: "Comparative Analysis of the Microbiome of PD flies and Control flies"
---

```{r setup, include=FALSE}

library(lme4)
library(multcomp)
library(dunn.test)
library(rcompanion)
library(tidyr)
library(ggplot2)
library(gridExtra)
library(reshape)
library(ade4)
library(grid)
library(gtable)
library(PMCMR)
library(multcompView)
library(splitstackshape)
library(readxl)
library(reshape2)
library(survival)
library(maps)
library(dplyr)
library(vegan)
library(cowplot)
library(stringr)
library(exactRankTests)
library(readr)
library(broom)


library(devtools)
library(usethis)
library(roxygen2)
library(learnr)


source(url("https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/16sscripts/make_taxon_plots_condense_2var_0.0.2.R"))
source(url("https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/16sscripts/g_legend.R"))
source(url("https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/16sscripts/pcoa_2var_0.0.2.R"))
source(url("https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/16sscripts/make_ancom_plots_0.0.2.R"))
source(url("https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/16sscripts/ancom_tests.R"))
source(url("https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/16sscripts/ancom_v2.0.R"))


knitr::opts_chunk$set(echo = FALSE)
```

## Introduction

### Welcome to the Parkinson Flies Microbiome Analysis Figures

In this tutorial you will see how all figures were created for the Parkinson Flies Microbiome Paper. Feel free to experiment with the tutorial by changing the taxon levels or other similar changes.

## Bash Work

### Using QIIME2


source activate qiime2-2019.7

cd ~/Park-Proj/Evan_stuff/parkFlies/Files/CURA_project

name=call19
mapper=call_mapper.tsv
seqsdir=reads/


*filter out wolbachia reads*
qiime taxa filter-table \
  --i-table table-$name"".qza \
  --i-taxonomy taxonomy-$name"".qza \
  --p-exclude Wolbachia \
  --o-filtered-table table-$name""_noW.qza
  
*filter out individual OTUs*
qiime feature-table filter-features\
  --i-table table-$name""_noW.qza \
  --m-metadata-file OTUs.tsv \
  --p-exclude-ids \
  --o-filtered-table table-$name""_feature.qza
  
*filter other unwanted contents*
qiime taxa filter-table \
  --i-table table-$name"".qza \
  --i-taxonomy taxonomy-$name"".qza \
  --p-exclude Mitochondira,Chloroplast,Archaea,Wolbachia \
  --o-filtered-table table-$name""_noOther.qza



bash ../16sscripts/standard_qiime3.sh $name""_noW $name UCLA_controls 164 $mapper "lab = 'UCLA' AND genotype_class = 'control'"

bash ../16sscripts/standard_qiime3.sh $name""_noW $name UCLA_mutants 164 $mapper "lab = 'UCLA' AND genotype_class = 'mutant'"

bash ../16sscripts/standard_qiime3.sh $name""_noW $name UCLA 164 $mapper "lab = 'UCLA'"



bash ../16sscripts/standard_qiime3.sh $name""_noW $name Stanford_controls 2974 $mapper "lab = 'Stanford' AND genotype_class = 'control'"

bash ../16sscripts/standard_qiime3.sh $name""_noW $name Stanford_mutants 2974 $mapper "lab = 'Stanford' AND genotype_class = 'mutant'"

bash ../16sscripts/standard_qiime3.sh $name""_noW $name Stanford 2974 $mapper "lab = 'Stanford'"



bash ../16sscripts/standard_qiime3.sh $name""_noW $name Pallanck_controls 7188 $mapper "lab = 'Pallanck'AND genotype_class = 'control'"

bash ../16sscripts/standard_qiime3.sh $name""_noW $name Pallanck_mutants 7188 $mapper "lab = 'Pallanck'AND genotype_class = 'mutant'"

bash ../16sscripts/standard_qiime3.sh $name""_noW $name Pallanck 7188 $mapper "lab = 'Pallanck'"



bash ../16sscripts/standard_qiime3.sh $name""_noW $name Bellen_controls 1565 $mapper "lab = 'Bellen' AND genotype_class = 'control'"

bash ../16sscripts/standard_qiime3.sh $name""_noW $name Bellen_mutants_174 1565 $mapper "lab = 'Bellen' AND genotype = 'iPLA2-VIA[Delta174]'"

bash ../16sscripts/standard_qiime3.sh $name""_noW $name Bellen_mutants_192 1565 $mapper "lab = 'Bellen' AND genotype = 'iPLA2-VIA[Delta192]'"

bash ../16sscripts/standard_qiime3.sh $name""_noW $name Bellen 1565 $mapper "lab = 'Bellen'"



bash ../16sscripts/standard_qiime3.sh $name""_feature $name All 200 $mapper "lab IN ('UCLA','Stanford','Bellen','Pallanck')"




bash ../16sscripts/standard_qiime3.sh $name""_noW $name iPLA174_noControl 1565 $mapper "genotype IN ('iPLA2-VIA[Delta174]')"

bash ../16sscripts/standard_qiime3.sh $name""_noW $name iPLA174 1565 $mapper "genotype IN ('iPLA2-VIA Control','iPLA2-VIA[Delta174]')"

bash ../16sscripts/standard_qiime3.sh $name""_noW $name Bloom_174 1565 $mapper "lab = 'Bloomington' AND genotype = 'iPLA2-VIA[Delta174]'"



bash ../16sscripts/standard_qiime3.sh $name""_noW $name iPLA192_noControl 1565 $mapper "genotype IN ('iPLA2-VIA[Delta192]')"

bash ../16sscripts/standard_qiime3.sh $name""_noW $name iPLA192 1565 $mapper "genotype IN ('iPLA2-VIA Control','iPLA2-VIA[Delta192]')"

bash ../16sscripts/standard_qiime3.sh $name""_noW $name Bloom_192 1565 $mapper "lab = 'Bloomington' AND genotype = 'iPLA2-VIA[Delta192]'"





















## Figure 1

*All sections of this figure were created separately and then brought together with grid arrange. This figure shows the various mutants in comparison to their controls.*

### Bellen Control Taxon Chart

```{r bellen_control, exercise=TRUE}

file_path <- "Bellen_controls"
urlpath <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/Bellen_controls/"
mapper_file <- url("https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/Bellen_controls/call_mapper.tsv")

var1 = "genotype"
taxonomic_level = "genus"
rpa_in_chart = 0.015
the_two_var = "onevar"
the_id = "X.SampleID"
plot_color = c("black","yellow","green", "brown","blue","red")
plot_order = c("Pasteurellales", "Enterococcus" ,"Lactobacillaceae", "Lactobacillus", "Acetobacter")
x_val = "genotype"
read_depth = 1565
map_path = "-call19"
legend_position = "none"

bellen_control_taxon <- make_taxon_plots_1var(urlpath = urlpath, file_path = file_path,	map_path = map_path, mapper_file = mapper_file, var1 = var1,	taxonomic_level = taxonomic_level,	rpa_in_chart = rpa_in_chart,	the_two_var = "onevar",	the_id = "X.SampleID",	plot_color = plot_color,	plot_order = plot_order,	x_val = x_val,read_depth = read_depth, legend_position = legend_position)

flies.labs <- c("iPLA2-VIA", "iPLA2-VIA[Delta174]", "iPLA2-VIA[Delta192]")
names(flies.labs) <- c("iPLA2-VIA Control", "iPLA2-VIA[Delta174]", "iPLA2-VIA[Delta192]")

bellen_control_taxon <- bellen_control_taxon + facet_grid(.~get(var1), drop = TRUE, space = "free", scales = "free", labeller = labeller(.cols = flies.labs)) +
  theme(
      strip.text.x = element_text(
        size = 30, color = "black", face = "bold"
        )
      )+ 
  xlab("Fly Samples") + 
  ylab("Fractional Abundance") +
  labs(fill = "Order Level Taxon")

plot(bellen_control_taxon)

```

### Bellen Mutant (Δ174) Taxon Chart

```{r bellen_mutant_174, exercise=TRUE}

file_path <- "Bellen_mutants_174"
urlpath <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/Bellen_mutants_174/"
mapper_file <- url("https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/Bellen_mutants_174/call_mapper.tsv")

var1 = "genotype"
taxonomic_level = "genus"
rpa_in_chart = 0.015
the_two_var = "onevar"
the_id = "X.SampleID"
plot_color = c("black","yellow", "brown","blue","red")
plot_order = c("Pasteurellales","Lactobacillaceae", "Lactobacillus", "Acetobacter")
x_val = "genotype"
read_depth = 1565
map_path = "-call19"
legend_position = "none"

bellen_mutant_174_taxon <- make_taxon_plots_1var(urlpath = urlpath, file_path = file_path,	map_path = map_path, mapper_file = mapper_file, var1 = var1,	taxonomic_level = taxonomic_level,	rpa_in_chart = rpa_in_chart,	the_two_var = "onevar",	the_id = "X.SampleID",	plot_color = plot_color,	plot_order = plot_order,	x_val = x_val,read_depth = read_depth, legend_position = legend_position)

flies.labs <- c("iPLA2-VIA", "iPLA2-VIA[Delta174]", "iPLA2-VIA[Delta192]")
names(flies.labs) <- c("iPLA2-VIA Control", "iPLA2-VIA[Delta174]", "iPLA2-VIA[Delta192]")

bellen_mutant_174_taxon <- bellen_mutant_174_taxon + facet_grid(.~get(var1), drop = TRUE, space = "free", scales = "free", labeller = labeller(.cols = flies.labs)) +
  theme(
      strip.text.x = element_text(
        size = 30, color = "black", face = "bold"
        )
      )+ 
  xlab("Fly Samples") + 
  ylab("Fractional Abundance") +
  labs(fill = "Order Level Taxon")

plot(bellen_mutant_174_taxon)

```

### Bellen Mutant (Δ192) Taxon Chart

```{r bellen_mutant_192, exercise=TRUE}

file_path <- "Bellen_mutants_192"
urlpath <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/Bellen_mutants_192/"
mapper_file <- url("https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/Bellen_mutants_192/call_mapper.tsv")

var1 = "genotype"
taxonomic_level = "genus"
rpa_in_chart = 0.03
the_two_var = "onevar"
the_id = "X.SampleID"
plot_color = c("black","yellow","green", "brown","blue","red")
plot_order = c("Pasteurellales", "Enterococcus" ,"Lactobacillaceae", "Lactobacillus", "Acetobacter")
x_val = "genotype"
read_depth = 1565
map_path = "-call19"
legend_position = "none"

bellen_mutant_192_taxon <- make_taxon_plots_1var(urlpath = urlpath, file_path = file_path,	map_path = map_path, mapper_file = mapper_file, var1 = var1,	taxonomic_level = taxonomic_level,	rpa_in_chart = rpa_in_chart,	the_two_var = "onevar",	the_id = "X.SampleID",	plot_color = plot_color,	plot_order = plot_order,	x_val = x_val,read_depth = read_depth, legend_position = legend_position)

flies.labs <- c("iPLA2-VIA", "iPLA2-VIA[Delta174]", "iPLA2-VIA[Delta192]")
names(flies.labs) <- c("iPLA2-VIA Control", "iPLA2-VIA[Delta174]", "iPLA2-VIA[Delta192]")

bellen_mutant_192_taxon <- bellen_mutant_192_taxon + facet_grid(.~get(var1), drop = TRUE, space = "free", scales = "free", labeller = labeller(.cols = flies.labs)) +
  theme(
      strip.text.x = element_text(
        size = 30, color = "black", face = "bold"
        )
      )+ 
  xlab("Fly Samples") + 
  ylab("Fractional Abundance") +
  labs(fill = "Order Level Taxon")

plot(bellen_mutant_192_taxon)

```

### Bellen PCoA

```{r Bellen_PCoA, exercise=TRUE}

file_path <- "Bellen"
urlpath <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/No_Wol/Bellen/"
mapper_file <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/No_Wol/Bellen/call_mapper.tsv"

map_file <- read.table(url(mapper_file),comment.char = "", header=T, fill=T, sep="\t") %>% mutate(X.SampleID = gsub(pattern = "-",replacement = "-",x = X.SampleID))

flies_weighted <- read.table(url(paste(urlpath,'core-metrics-results-',file_path,'/weighted_unifrac_distance_matrix/distance-matrix.tsv',sep="")), header=T, sep="\t")
flies_weighted_dm <- as.dist(flies_weighted[,2:dim(flies_weighted)[2]])
flies_unifrac_table <- flies_weighted %>% dplyr::select(X) %>% left_join(map_file, by=c("X"="X.SampleID")) %>% droplevels()

table <- adonis2(flies_weighted_dm ~ genotype_class * vial_model, flies_unifrac_table, permutations=1000)
table <- tidy(round(table,2))
print(table)

Bellen_geno_p <- table %>%
  filter(term == "genotype_class") %>%
  pull(p.value) %>%
  round(3)
Bellen_geno_r <- table %>%
  filter(term == "genotype_class") %>%
  pull(R2) %>%
  round(3)

var1 = "genotype"
var1_colors <-c("blue","red", "red")
legend_position = "none"
folder_name = paste0(file_path, "/weighted_unifrac")
title_name_add <- "weighted Unifrac"
labs = c("iPLA2-VIA (Control)", "iPLA2-VIA[Delta174] (Mutant)", "iPLA2-VIA[Delta192] (Mutant)")


Bellen_w_PCOA <- pcoa_1var(urlpath = urlpath,folder_name=folder_name, mapper_file = mapper_file, var1=var1, var1_colors=var1_colors, title_name_add=title_name_add, legend_position = legend_position)
Bellen_w_PCOA <- Bellen_w_PCOA + scale_color_manual(name="Genotype", labels=labs, values=var1_colors) + 
  scale_fill_manual(name="Genotype", labels=labs, values=var1_colors) +
  ggtitle(paste0("R2=", Bellen_geno_r, ", p=", Bellen_geno_p)) +
  theme(text = element_text(size = 30), axis.title = element_text(size = 30), plot.title = element_text(size = 30, hjust = 1)) +
  geom_point(size = 5) +
  stat_ellipse(aes(x=V2, y=V3,group= get(var1)), lwd = 1, level = .9, show.legend = F, type = "t", geom = "polygon", alpha = 0, inherit.aes=T)

plot(Bellen_w_PCOA)
```

### Pallanck Control Taxon Chart

```{r pallanck_control, exercise=TRUE}

file_path <- "Pallanck_controls"
urlpath <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/Pallanck_controls/"
mapper_file <- url("https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/Pallanck_controls/call_mapper.tsv")

var1 = "genotype"
taxonomic_level = "genus"
rpa_in_chart = 0.05
the_two_var = "onevar"
the_id = "X.SampleID"
plot_color = c("black","brown","green","red")
plot_order = c("Lactobacillaceae","Enterococcus", "Acetobacter")
x_val = "genotype"
read_depth = 7188
map_path = "-call19"
legend_position = "none"

pallanck_control_taxon <- make_taxon_plots_1var(urlpath = urlpath, file_path = file_path,	map_path = map_path, mapper_file = mapper_file, var1 = var1, taxonomic_level = taxonomic_level,	rpa_in_chart = rpa_in_chart,	the_two_var = "onevar",	the_id = "X.SampleID",	plot_color = plot_color,	plot_order = plot_order,	x_val = x_val,read_depth = read_depth, legend_position = legend_position)

flies.labs <- c("park25", "parkrvA")
names(flies.labs) <- c("park25", "parkrvA (Control)")
pallanck_control_taxon <- pallanck_control_taxon + facet_grid(.~get(var1), drop = TRUE, space = "free", scales = "free", labeller = labeller(.cols = flies.labs)) +
  theme(
      strip.text.x = element_text(
        size = 30, color = "black", face = "bold"
        )
      )+ 
  xlab("Fly Samples") + 
  ylab("Fractional Abundance") +
  labs(fill = "Order Level Taxon")

plot(pallanck_control_taxon)

```

### Pallanck Mutant Taxon Chart

```{r pallanck_mutant, exercise=TRUE}

file_path <- "Pallanck_mutants"
urlpath <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/Pallanck_mutants/"
mapper_file <- url("https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/Pallanck_mutants/call_mapper.tsv")

var1 = "genotype"
taxonomic_level = "genus"
rpa_in_chart = 0.015
the_two_var = "onevar"
the_id = "X.SampleID"
plot_color = c("black","brown","red")
plot_order = c("Lactobacillaceae", "Acetobacter")
x_val = "genotype"
read_depth = 7188
map_path = "-call19"
legend_position = "none"

pallanck_mutant_taxon <- make_taxon_plots_1var(urlpath = urlpath, file_path = file_path,	map_path = map_path, mapper_file = mapper_file, var1 = var1, taxonomic_level = taxonomic_level,	rpa_in_chart = rpa_in_chart,	the_two_var = "onevar",	the_id = "X.SampleID",	plot_color = plot_color,	plot_order = plot_order,	x_val = x_val,read_depth = read_depth, legend_position = legend_position)

flies.labs <- c("park25", "parkrvA")
names(flies.labs) <- c("park25", "parkrvA (Control)")
pallanck_mutant_taxon <- pallanck_mutant_taxon + facet_grid(.~get(var1), drop = TRUE, space = "free", scales = "free", labeller = labeller(.cols = flies.labs)) +
  theme(
      strip.text.x = element_text(
        size = 30, color = "black", face = "bold"
        )
      )+ 
  xlab("Fly Samples") + 
  ylab("Fractional Abundance") +
  labs(fill = "Order Level Taxon")

plot(pallanck_mutant_taxon)

```

### Pallanck PCoA

```{r Pallanck_PCoA, exercise=TRUE}

file_path <- "Pallanck"
urlpath <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/No_Wol/Pallanck/"
mapper_file <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/No_Wol/Pallanck/call_mapper.tsv"

map_file <- read.table(url(mapper_file),comment.char = "", header=T, fill=T, sep="\t") %>% mutate(X.SampleID = gsub(pattern = "-",replacement = "-",x = X.SampleID))

flies_weighted <- read.table(url(paste(urlpath,'core-metrics-results-',file_path,'/weighted_unifrac_distance_matrix/distance-matrix.tsv',sep="")), header=T, sep="\t")
flies_weighted_dm <- as.dist(flies_weighted[,2:dim(flies_weighted)[2]])
flies_unifrac_table <- flies_weighted %>% dplyr::select(X) %>% left_join(map_file, by=c("X"="X.SampleID")) %>% droplevels()

table <- adonis2(flies_weighted_dm ~ genotype * vial_model, flies_unifrac_table, permutations=1000)
table <- tidy(round(table,2))
print(table)

Pallanck_geno_p <- table %>%
  filter(term == "genotype") %>%
  pull(p.value) %>%
  round(3)
Pallanck_geno_r <- table %>%
  filter(term == "genotype") %>%
  pull(R2) %>%
  round(3)

var1 = "genotype_class"
var1_colors <-c("red","blue")
legend_position = "none"
folder_name = paste0(file_path, "/weighted_unifrac")
title_name_add <- "weighted Unifrac"
labs <- c("Mutant", "Control")


Pallanck_w_PCOA <- pcoa_1var(urlpath = urlpath, folder_name=folder_name, mapper_file = mapper_file, var1=var1, var1_colors=var1_colors, title_name_add=title_name_add, legend_position = legend_position)
Pallanck_w_PCOA <- Pallanck_w_PCOA + scale_color_manual(name="Genotype", labels=labs, values=var1_colors) + 
  scale_fill_manual(name="Genotype", labels=labs, values=var1_colors) +
  ggtitle(paste0("R2=", Pallanck_geno_r,", p=", Pallanck_geno_p)) +
  theme(text = element_text(size = 30), axis.title = element_text(size = 30), plot.title = element_text(size = 30, hjust = 1)) +
  geom_point(size = 5) +
  stat_ellipse(aes(x=V2, y=V3,group= get(var1)), lwd = 1, level = .9, show.legend = F, type = "t", geom = "polygon", alpha = 0, inherit.aes=T)

plot(Pallanck_w_PCOA)
```

### Stanford Control Taxon Chart

```{r stanford_controls, exercise=TRUE}

file_path <- "Stanford_controls"
urlpath <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/Stanford_controls/"
mapper_file <- url("https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/Stanford_controls/call_mapper.tsv")

var1 = "genotype"
taxonomic_level = "genus"
rpa_in_chart = 0.002
the_two_var = "onevar"
the_id = "X.SampleID"
plot_color = c("black", "blue","red")
plot_order = c( "Lactobacillus", "Acetobacter")
x_val = "genotype"
read_depth = 2974
map_path = "-call19"
legend_position = "none"

stanford_control_taxon <- make_taxon_plots_1var(urlpath = urlpath, file_path = file_path,	map_path = map_path, mapper_file = mapper_file, var1 = var1, taxonomic_level = taxonomic_level,	rpa_in_chart = rpa_in_chart,	the_two_var = "onevar",	the_id = "X.SampleID",	plot_color = plot_color,	plot_order = plot_order,	x_val = x_val,read_depth = read_depth, legend_position = legend_position)

flies.labs <- c("pink1 B9 FM6", "pink1 RV FM6")
names(flies.labs) <- c("pink1 B9 FM6 Mutants", "pink1 RV FM6 (Control)")
stanford_control_taxon <- stanford_control_taxon + facet_grid(.~get(var1), drop = TRUE, space = "free", scales = "free", labeller = labeller(.cols = flies.labs)) +
  theme(
      strip.text.x = element_text(
        size = 30, color = "black", face = "bold"
        )
      )+ 
  xlab("Fly Samples") + 
  ylab("Fractional Abundance") +
  labs(fill = "Order Level Taxon")

plot(stanford_control_taxon)

```

### Stanford Mutant Taxon Chart

```{r stanford_mutants, exercise=TRUE}

file_path <- "Stanford_mutants"
urlpath <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/Stanford_mutants/"
mapper_file <- url("https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/Stanford_mutants/call_mapper.tsv")

var1 = "genotype"
taxonomic_level = "genus"
rpa_in_chart = 0.004
the_two_var = "onevar"
the_id = "X.SampleID"
plot_color = c("black", "brown", "blue","red")
plot_order = c( "Lactobacillaceae", "Lactobacillus", "Acetobacter")
x_val = "genotype"
read_depth = 2974
map_path = "-call19"
legend_position = "none"

stanford_mutant_taxon <- make_taxon_plots_1var(urlpath = urlpath, file_path = file_path,	map_path = map_path, mapper_file = mapper_file, var1 = var1, taxonomic_level = taxonomic_level,	rpa_in_chart = rpa_in_chart,	the_two_var = "onevar",	the_id = "X.SampleID",	plot_color = plot_color,	plot_order = plot_order,	x_val = x_val,read_depth = read_depth, legend_position = legend_position)

flies.labs <- c("pink1 B9 FM6", "pink1 RV FM6")
names(flies.labs) <- c("pink1 B9 FM6 Mutants", "pink1 RV FM6 (Control)")
stanford_mutant_taxon <- stanford_mutant_taxon + facet_grid(.~get(var1), drop = TRUE, space = "free", scales = "free", labeller = labeller(.cols = flies.labs)) +
  theme(
      strip.text.x = element_text(
        size = 30, color = "black", face = "bold"
        )
      )+ 
  xlab("Fly Samples") + 
  ylab("Fractional Abundance") +
  labs(fill = "Order Level Taxon")

plot(stanford_mutant_taxon)

```

### Stanford PCoA

```{r Stanford_PCoA, exercise=TRUE}

file_path <- "Stanford"
urlpath <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/No_Wol/Stanford/"
mapper_file <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/No_Wol/Stanford/call_mapper.tsv"

map_file <- read.table(url(mapper_file),comment.char = "", header=T, fill=T, sep="\t") %>% mutate(X.SampleID = gsub(pattern = "-",replacement = "-",x = X.SampleID))

flies_weighted <- read.table(url(paste(urlpath,'core-metrics-results-',file_path,'/weighted_unifrac_distance_matrix/distance-matrix.tsv',sep="")), header=T, sep="\t")
flies_weighted_dm <- as.dist(flies_weighted[,2:dim(flies_weighted)[2]])
flies_unifrac_table <- flies_weighted %>% dplyr::select(X) %>% left_join(map_file, by=c("X"="X.SampleID")) %>% droplevels()

table <- adonis2(flies_weighted_dm ~ genotype * vial_model, flies_unifrac_table, permutations=1000)
table <- tidy(round(table,2))
print(table)

Stanford_geno_p <- table %>%
  filter(term == "genotype") %>%
  pull(p.value) %>%
  round(3)
Stanford_geno_r <- table %>%
  filter(term == "genotype") %>%
  pull(R2) %>%
  round(3)

var1 = "genotype_class"
var1_colors <-c("red","blue")
legend_position = "none"
folder_name = paste0(file_path, "/weighted_unifrac")
title_name_add <- "weighted Unifrac"
labs <- c("Control", "Mutant")


Stanford_w_PCOA <- pcoa_1var(urlpath = urlpath, folder_name=folder_name, mapper_file = mapper_file, var1=var1, var1_colors=var1_colors, title_name_add=title_name_add, legend_position = legend_position)
Stanford_w_PCOA <- Stanford_w_PCOA + scale_color_manual(name="Genotype", labels=labs, values=var1_colors) + 
  scale_fill_manual(name="Genotype", labels=labs, values=var1_colors) +
  ggtitle(paste0("R2=", Stanford_geno_r,", p=", Stanford_geno_p)) +
  theme(text = element_text(size = 30), axis.title = element_text(size = 30), plot.title = element_text(size = 30, hjust = 1)) +
  geom_point(size = 5) +
  stat_ellipse(aes(x=V2, y=V3,group= get(var1)), lwd = 1, level = .9, show.legend = F, type = "t", geom = "polygon", alpha = 0, inherit.aes=T)

plot(Stanford_w_PCOA)
```

### UCLA Control Taxon Chart

```{r ucla_control, exercise=TRUE}

file_path <- "UCLA_controls"
urlpath <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/UCLA_controls/"
mapper_file <- url("https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/UCLA_controls/call_mapper.tsv")

var1 = "genotype"
taxonomic_level = "genus"
rpa_in_chart = 0.04 ### relative percent abundance in chart -> taxa that are at least this abundant overall will show up
the_two_var = "onevar" ## ignore
the_id = "X.SampleID" ## ignore
plot_color = c("black","brown","blue","red")
plot_order = c("Lactobacillaceae", "Lactobacillus","Acetobacter")
x_val = "genotype"
read_depth = 164
map_path = "-call19" ## name of the taxonomy file; write instructions on how to create a taxonomy file - taxonomy_forR.csv
legend_position = "none"

ucla_control_taxon <- make_taxon_plots_1var(urlpath = urlpath, file_path = file_path,	map_path = map_path, mapper_file = mapper_file, var1 = var1, taxonomic_level = taxonomic_level,	rpa_in_chart = rpa_in_chart,	the_two_var = "onevar",	the_id = "X.SampleID",	plot_color = plot_color,	plot_order = plot_order,	x_val = x_val,read_depth = read_depth, legend_position = legend_position)

flies.labs <- c("PG", "pink15")
names(flies.labs) <- c("PG (Control)", "pink15")
ucla_control_taxon <- ucla_control_taxon + facet_grid(.~get(var1), drop = TRUE, space = "free", scales = "free", labeller = labeller(.cols = flies.labs)) +
  theme(
      strip.text.x = element_text(
        size = 30, color = "black", face = "bold"
        )
      )+ 
  xlab("Fly Samples") + 
  ylab("Fractional Abundance") +
  labs(fill = "Order Level Taxon")

plot(ucla_control_taxon)

```

### UCLA Mutant Taxon Chart

```{r ucla_mutant, exercise=TRUE}

file_path <- "UCLA_mutants"
urlpath <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/UCLA_mutants/"
mapper_file <- url("https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/UCLA_mutants/call_mapper.tsv")

var1 = "genotype"
taxonomic_level = "genus"
rpa_in_chart = 0.05 ### relative percent abundance in chart -> taxa that are at least this abundant overall will show up
the_two_var = "onevar" ## ignore
the_id = "X.SampleID" ## ignore
plot_color = c("black","green","brown")
plot_order = c("Enterococcus", "Lactobacillaceae")
x_val = "genotype"
read_depth = 164
map_path = "-call19" ## name of the taxonomy file; write instructions on how to create a taxonomy file - taxonomy_forR.csv
legend_position = "none"

ucla_mutant_taxon <- make_taxon_plots_1var(urlpath = urlpath, file_path = file_path,	map_path = map_path, mapper_file = mapper_file, var1 = var1, taxonomic_level = taxonomic_level,	rpa_in_chart = rpa_in_chart,	the_two_var = "onevar",	the_id = "X.SampleID",	plot_color = plot_color,	plot_order = plot_order,	x_val = x_val,read_depth = read_depth, legend_position = legend_position)

flies.labs <- c("PG", "pink15")
names(flies.labs) <- c("PG (Control)", "pink15")
ucla_mutant_taxon <- ucla_mutant_taxon + facet_grid(.~get(var1), drop = TRUE, space = "free", scales = "free", labeller = labeller(.cols = flies.labs)) +
  theme(
      strip.text.x = element_text(
        size = 30, color = "black", face = "bold"
        )
      )+ 
  xlab("Fly Samples") + 
  ylab("Fractional Abundance") +
  labs(fill = "Order Level Taxon")

plot(ucla_mutant_taxon)

```

### UCLA PCoA

```{r UCLA_PCoA, exercise=TRUE}

file_path <- "UCLA"
urlpath <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/No_Wol/UCLA/"
mapper_file <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/No_Wol/UCLA/call_mapper.tsv"

map_file <- read.table(url(mapper_file),comment.char = "", header=T, fill=T, sep="\t") %>% mutate(X.SampleID = gsub(pattern = "-",replacement = "-",x = X.SampleID))

flies_weighted <- read.table(url(paste(urlpath,'core-metrics-results-',file_path,'/weighted_unifrac_distance_matrix/distance-matrix.tsv',sep="")), header=T, sep="\t")
flies_weighted_dm <- as.dist(flies_weighted[,2:dim(flies_weighted)[2]])
flies_unifrac_table <- flies_weighted %>% dplyr::select(X) %>% left_join(map_file, by=c("X"="X.SampleID")) %>% droplevels()

table <- adonis2(flies_weighted_dm ~ genotype * vial_model, flies_unifrac_table, permutations=1000)
table <- tidy(round(table,2))
print(table)

UCLA_geno_p <- table %>%
  filter(term == "genotype") %>%
  pull(p.value) %>%
  round(3)
UCLA_geno_r <- table %>%
  filter(term == "genotype") %>%
  pull(R2) %>%
  round(3)

var1 = "genotype_class"
var1_colors <-c("blue", "red")
legend_position = "none"
folder_name = paste0(file_path, "/weighted_unifrac")
title_name_add <- "weighted Unifrac"
labs <- c("Control", "Mutant")


UCLA_w_PCOA <- pcoa_1var(urlpath = urlpath, folder_name=folder_name, mapper_file = mapper_file, var1=var1, var1_colors=var1_colors, title_name_add=title_name_add, legend_position = legend_position)
UCLA_w_PCOA <- UCLA_w_PCOA + scale_color_manual(name="Genotype", labels=labs, values=var1_colors) + 
  scale_fill_manual(name="Genotype", labels=labs, values=var1_colors) +
  ggtitle(paste0("R2=", UCLA_geno_r,", p=", UCLA_geno_p)) +
  theme(text = element_text(size = 30), axis.title = element_text(size = 30), plot.title = element_text(size = 30, hjust = 1)) +
  geom_point(size = 5) +
  stat_ellipse(aes(x=V2, y=V3,group= get(var1)), lwd = 1, level = .9, show.legend = F, type = "t", geom = "polygon", alpha = 0, inherit.aes=T)

plot(UCLA_w_PCOA)
```

### Figure Microbe and PCoA Key

```{r figure1_microbe_key, exercise=TRUE}

file_path <- "All"
urlpath <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/No_Wol/All/"
mapper_file <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/No_Wol/All/call_mapper.tsv"

var1 = "genotype"
taxonomic_level = "genus"
rpa_in_chart = 0.015
the_two_var = "onevar"
the_id = "X.SampleID"
plot_color = c("black","yellow", "green", "brown","blue", "red")
plot_order = c("Pasteurellales", "Enterococcus", "Lactobacillaceae", "Lactobacillus","Acetobacter")
x_val = "genotype_class"
read_depth = 2000
map_path = "-call19"
legend_position = "bottom"

All_labs <- make_taxon_plots_1var(urlpath = urlpath, file_path = file_path,	map_path = map_path, mapper_file = mapper_file, var1 = var1,	taxonomic_level = taxonomic_level,	rpa_in_chart = rpa_in_chart,	the_two_var = "onevar",	the_id = "X.SampleID",	plot_color = plot_color,	plot_order = plot_order,	x_val = x_val,read_depth = read_depth, legend_position = legend_position)


flies.labs <- c("Control", "Mutant")
names(flies.labs) <- c("control", "mutant")

All_labs <- All_labs + facet_grid(.~get(var1), drop = TRUE, space = "free", scales = "free", labeller = labeller(.cols = flies.labs)) +
  theme(
      strip.text.x = element_text(
        size = 35, color = "black", face = "bold"
        ),
      axis.title=element_text(size=35),
                   title=element_text(size=40),
      legend.text = element_text(size = 28),
      legend.key.size = unit(1, 'cm'),
      )+ 
  xlab("Fly Samples") + 
  ylab("Fractional Abundance") +
  labs(fill = "Genus", title = "Overall Taxon")

legend_position = "bottom"
plot_fig1legend <- g_legend(All_labs + theme(legend.title = NULL))

plot_fig1legend + theme(legend.text = element_text(size = 28),
      legend.key.size = unit(2, 'cm'))

plot(plot_fig1legend)




folder_name = paste0(file_path, "/weighted_unifrac")
title_name_add <- "weighted Unifrac"
var1 = "genotype_class"
var1_colors <-c("blue", "red")
labs <- c("Control", "Mutant")

All_lab_wPCoA <- pcoa_1var(urlpath = urlpath, folder_name=folder_name, mapper_file = mapper_file, var1=var1, var1_colors=var1_colors, title_name_add=title_name_add, legend_position = legend_position)
All_lab_wPCoA <- All_lab_wPCoA + 
  scale_color_manual(name="Genotype", labels=labs, values=var1_colors) + 
  scale_fill_manual(name="Genotype", labels=labs, values=var1_colors) +
  theme(text = element_text(size = 35), axis.title = element_text(size = 35), plot.title = element_text(size = 35)) +
  geom_point(size = 10) +
  stat_ellipse(aes(x=V2, y=V3,group= get(var1)), lwd = 4, level = .9, show.legend = F, type = "t", geom = "polygon", alpha = 0, inherit.aes=T)

legend_position = "bottom"
plot_wuwlegend <- g_legend(All_lab_wPCoA)
plot(plot_wuwlegend)

```

### Label Creation

```{r labels, exercise=TRUE}
control = textGrob("Control", gp = gpar(fontsize = 45, fontface = "bold"))
mutant = textGrob("Mutant", gp = gpar(fontsize = 45, fontface = "bold"))
UCLA = textGrob("UCLA", gp = gpar(fontsize = 45, fontface = "bold"), rot = 90)
Stanford = textGrob("Stanford", gp = gpar(fontsize = 45, fontface = "bold"), rot = 90)
Pallanck = textGrob("Pallanck", gp = gpar(fontsize = 45, fontface = "bold"), rot = 90)
Bellen = textGrob("Bellen", gp = gpar(fontsize = 45, fontface = "bold"), rot = 90)
var1 = "genotype"
```

### Figure Creation

```{r figure1, exercise=TRUE}

jpeg(h=1300, w=1800, "Figure_1.jpg", units = "px", quality = 100, pointsize = 1)
  grid.arrange(
		ucla_control_taxon, ucla_mutant_taxon, stanford_control_taxon, stanford_mutant_taxon,
		pallanck_control_taxon, pallanck_mutant_taxon, bellen_control_taxon, bellen_mutant_174_taxon,
		control, mutant, UCLA, Stanford, Pallanck, Bellen, bellen_mutant_192_taxon,UCLA_w_PCOA + theme(plot.title = element_text(size = 20)), Stanford_w_PCOA + theme(plot.title = element_text(size = 20)),Bellen_w_PCOA + theme(plot.title = element_text(size = 20)),Pallanck_w_PCOA + theme(plot.title = element_text(size = 20)),plot_fig1legend,plot_wuwlegend,
		widths = c(.25,1.5,1.5,1.5,.2,1.5),	
   	heights = c(.25,1,1,1,1,.5),
		layout_matrix=rbind(
		  c(NA,9,10,10,NA,NA),
		  c(14,7,8,15,NA,18),
		  c(13,5,6,NA,NA,19),
		  c(12,3,4,NA,NA,17),
			c(11,1,2,NA,NA,16),
			c(NA,20,20,20,NA,21))
		)

dev.off()

```

## Table 1

*This is a table showing the PERMANOVA analysis of all mutants versus all controls using the Unweighted Unifrac, Weighted Unifrac, and Bray Curtis Distance Matrices.*

### Unweighted Unifrac

```{r all_unweighted, exercise=TRUE}

file_path <- "All"
urlpath <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/No_Wol/All/"
mapper_file <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/No_Wol/All/call_mapper.tsv"

flies_unweighted <- read.table(url(paste(urlpath,'core-metrics-results-',file_path,'/unweighted_unifrac_distance_matrix/distance-matrix.tsv',sep="")), header=T, sep="\t") %>% mutate(X=as.character(X))
flies_unweighted_dm <- as.dist(flies_unweighted[,2:dim(flies_unweighted)[2]])

map_file <- read.table(mapper_file,comment.char = "", header=T, fill=T, sep="\t") %>% mutate(X.SampleID = gsub(pattern = "-",replacement = "-",x = X.SampleID))
flies_unifrac_table <- flies_unweighted %>% select(X) %>% left_join(map_file, by=c("X"="X.SampleID")) %>% droplevels()

table <- adonis2(flies_unweighted_dm ~ lab * genotype_class * vial_model, flies_unifrac_table, permutations=1000)
table <- tidy(round(table, 2))

print(table)

```

### Weighted Unifrac

```{r all_weighted, exercise=TRUE}

file_path <- "All"
urlpath <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/No_Wol/All/"
mapper_file <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/No_Wol/All/call_mapper.tsv"

flies_weighted <- read.table(url(paste(urlpath,'core-metrics-results-',file_path,'/weighted_unifrac_distance_matrix/distance-matrix.tsv',sep="")), header=T, sep="\t")
flies_weighted_dm <- as.dist(flies_weighted[,2:dim(flies_weighted)[2]])

map_file <- read.table(mapper_file,comment.char = "", header=T, fill=T, sep="\t") %>% mutate(X.SampleID = gsub(pattern = "-",replacement = "-",x = X.SampleID))
flies_unifrac_table <- flies_weighted %>% select(X) %>% left_join(map_file, by=c("X"="X.SampleID")) %>% droplevels()

table <- adonis2(flies_weighted_dm ~ lab * genotype_class * vial_model, flies_unifrac_table, permutations=1000)
table <- tidy(round(table, 2))

print(table)

```

### Bray Curtis

```{r all_brayCurtis, exercise=TRUE}

file_path <- "All"
urlpath <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/No_Wol/All/"
mapper_file <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/No_Wol/All/call_mapper.tsv"

flies_bc <- read.table(url(paste(urlpath,'core-metrics-results-',file_path,'/bray_curtis_distance_matrix/distance-matrix.tsv',sep="")), header=T, sep="\t")
flies_bc_dm <- as.dist(flies_bc[,2:dim(flies_bc)[2]])

map_file <- read.table(mapper_file,comment.char = "", header=T, fill=T, sep="\t") %>% mutate(X.SampleID = gsub(pattern = "-",replacement = "-",x = X.SampleID))
flies_unifrac_table <- flies_bc %>% select(X) %>% left_join(map_file, by=c("X"="X.SampleID")) %>% droplevels()

table <- adonis2(flies_bc_dm ~ lab * genotype_class * vial_model, flies_unifrac_table, permutations=1000)
table <- tidy(round(table, 2))

print(table)

```

## Figure 2

*All sections of this figure were created separately and then brought together with grid arrange. This figure shows the pairwise comparison between the Bellen iPLA2-VIA[Delta174] mutants and the Bloomington iPLA2-VIA[Delta174] mutants.*

### Bellen Δ174 Taxon Chart

```{r bellen_174, exercise=TRUE}

file_path <- "Bellen_mutants_174"
urlpath <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/Bellen_mutants_174/"
mapper_file <- url("https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/Bellen_mutants_174/call_mapper.tsv")

var1 = "genotype"
taxonomic_level = "genus"
rpa_in_chart = 0.015
the_two_var = "onevar"
the_id = "X.SampleID"
plot_color = c("black","yellow", "brown","blue","red")
plot_order = c("Pasteurellales","Lactobacillaceae", "Lactobacillus", "Acetobacter")
x_val = "genotype"
read_depth = 1565
map_path = "-call19"
legend_position = "none"

bellen_mutant_174_taxon <- make_taxon_plots_1var(urlpath = urlpath, file_path = file_path,	map_path = map_path, mapper_file = mapper_file, var1 = var1,	taxonomic_level = taxonomic_level,	rpa_in_chart = rpa_in_chart,	the_two_var = "onevar",	the_id = "X.SampleID",	plot_color = plot_color,	plot_order = plot_order,	x_val = x_val,read_depth = read_depth, legend_position = legend_position)

flies.labs <- c("iPLA2-VIA", "iPLA2-VIA[Delta174] (Bellen)", "iPLA2-VIA[Delta192]")
names(flies.labs) <- c("iPLA2-VIA Control", "iPLA2-VIA[Delta174]", "iPLA2-VIA[Delta192]")

bellen_mutant_174_taxon <- bellen_mutant_174_taxon + facet_grid(.~get(var1), drop = TRUE, space = "free", scales = "free", labeller = labeller(.cols = flies.labs)) +
  theme(
      strip.text.x = element_text(
        size = 30, color = "black", face = "bold"
        )
      )+ 
  xlab("Fly Samples") + 
  ylab("Fractional Abundance") +
  labs(fill = "Order Level Taxon")

plot(bellen_mutant_174_taxon)

```

### Bloomington Δ174 Taxon Chart

```{r Bloomington_174, exercise=TRUE}

file_path <- "Bloom_174"
urlpath <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/Bloom_174/"
mapper_file <- url("https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/Bloom_174/call_mapper.tsv")

var1 = "genotype"
taxonomic_level = "genus"
rpa_in_chart = 0.005
the_two_var = "onevar"
the_id = "X.SampleID"
plot_color = c("black","brown","blue","red")
plot_order = c("Lactobacillaceae", "Lactobacillus", "Acetobacter")
x_val = "genotype"
read_depth = 1565
map_path = "-call19"
legend_position = "none"

bloom_174_taxon <- make_taxon_plots_1var(urlpath, file_path = file_path,	map_path = map_path, mapper_file = mapper_file, var1 = var1,	taxonomic_level = taxonomic_level,	rpa_in_chart = rpa_in_chart,	the_two_var = "onevar",	the_id = "X.SampleID",	plot_color = plot_color,	plot_order = plot_order,	x_val = x_val,read_depth = read_depth, legend_position = legend_position)

flies.labs <- c("iPLA2-VIA", "iPLA2-VIA[Delta174] (Bloomington)", "iPLA2-VIA[Delta192]")
names(flies.labs) <- c("iPLA2-VIA Control", "iPLA2-VIA[Delta174]", "iPLA2-VIA[Delta192]")

bloom_174_taxon <- bloom_174_taxon + facet_grid(.~get(var1), drop = TRUE, space = "free", scales = "free", labeller = labeller(.cols = flies.labs)) +
  theme(
      strip.text.x = element_text(
        size = 30, color = "black", face = "bold"
        )
      )+ 
  xlab("Fly Samples") + 
  ylab("Fractional Abundance") +
  labs(fill = "Order Level Taxon")

plot(bloom_174_taxon)

```

### Δ174 PERMANOVA and PCoA Comparison

```{r 174_perm_pcoa, exercise=TRUE}

file_path <- "iPLA174_noControl"
urlpath <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/iPLA174_noControl/"
mapper_file <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/iPLA174_noControl/call_mapper.tsv"

map_file <- read.table(mapper_file,comment.char = "", header=T, fill=T, sep="\t") %>% mutate(X.SampleID = gsub(pattern = "-",replacement = "-",x = X.SampleID))

flies_weighted <- read.table(url(paste(urlpath,'core-metrics-results-',file_path,'/weighted_unifrac_distance_matrix/distance-matrix.tsv',sep="")), header=T, sep="\t")
flies_weighted_dm <- as.dist(flies_weighted[,2:dim(flies_weighted)[2]])
flies_unifrac_table <- flies_weighted %>% dplyr::select(X) %>% left_join(map_file, by=c("X"="X.SampleID")) %>% droplevels()

table <- adonis2(flies_weighted_dm ~ lab * vial_model, flies_unifrac_table, permutations=1000)
perm_174_noControl <- tableGrob(round(table, 2), theme=ttheme_minimal())
table <- tidy(round(table,2))
print("PERMANOVA analysis:")
print(table)

i174_lab_p <- table %>%
  filter(term == "lab") %>%
  pull(p.value) %>%
  round(3)
i174_lab_r <- table %>%
  filter(term == "lab") %>%
  pull(R2) %>%
  round(3)

var1 = "lab"
var1_colors <-c("blue","red")
legend_position = "bottom"
labs = c("Bellen", "Bloomington")
folder_name = paste0(file_path, "/weighted_unifrac")
title_name_add <- "weighted Unifrac"

PCoA_174_noControl <- pcoa_1var(urlpath = urlpath, folder_name=folder_name, mapper_file = mapper_file, var1=var1, var1_colors=var1_colors, title_name_add=title_name_add, legend_position = legend_position)
PCoA_174_noControl <- PCoA_174_noControl + scale_color_manual(name="Lab", labels=labs, values=var1_colors) + 
  scale_fill_manual(name="Lab", labels=labs, values=var1_colors) +
  ggtitle(paste0("R2=", i174_lab_r,", p=", i174_lab_p)) +
  theme(text = element_text(size = 30), axis.title = element_text(size = 30), plot.title = element_text(size = 30, hjust = 1)) +
  geom_point(size = 5) +
  stat_ellipse(aes(x=V2, y=V3,group= get(var1)), lwd = 1, level = .9, show.legend = F, type = "t", geom = "polygon", alpha = 0, inherit.aes=T)

plot(PCoA_174_noControl)

```

### Figure Creation

``` {r figure2, exercise=TRUE}

jpeg(h=800, w=1200, "Figure_2.jpg", units = "px", quality = 100, pointsize = 1)
	grid.arrange(
		bellen_mutant_174_taxon + labs(tag = "A") + theme(plot.tag = element_text(size = 45)), bloom_174_taxon + labs(tag = "B") + theme(plot.tag = element_text(size = 45)), PCoA_174_noControl + labs(tag = "C") + theme(plot.tag = element_text(size = 45)), perm_174_noControl ,
		widths = c(1.5,1.5),	
   	heights = c(1, 1),
		layout_matrix=rbind(
		  c(1,2),
		  c(3,4))
		)
dev.off()

```

## Figure 3

*All sections of this figure were created separately and then brought together with grid arrange. This figure shows the pairwise comparison between the Bellen iPLA2-VIA[Delta192] mutants and the Bloomington iPLA2-VIA[Delta192] mutants.*

### Bellen Δ192 Taxon Chart

```{r bellen_192, exercise=TRUE}

file_path <- "Bellen_mutants_192"
urlpath <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/Bellen_mutants_192/"
mapper_file <- url("https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/Bellen_mutants_192/call_mapper.tsv")

var1 = "genotype"
taxonomic_level = "genus"
rpa_in_chart = 0.03
the_two_var = "onevar"
the_id = "X.SampleID"
plot_color = c("black","yellow","green", "brown","blue","red")
plot_order = c("Pasteurellales", "Enterococcus" ,"Lactobacillaceae", "Lactobacillus", "Acetobacter")
x_val = "genotype"
read_depth = 1565
map_path = "-call19"
legend_position = "none"

bellen_mutant_192_taxon <- make_taxon_plots_1var(urlpath = urlpath, file_path = file_path,	map_path = map_path, mapper_file = mapper_file, var1 = var1,	taxonomic_level = taxonomic_level,	rpa_in_chart = rpa_in_chart,	the_two_var = "onevar",	the_id = "X.SampleID",	plot_color = plot_color,	plot_order = plot_order,	x_val = x_val,read_depth = read_depth, legend_position = legend_position)

flies.labs <- c("iPLA2-VIA", "iPLA2-VIA[Delta174]", "iPLA2-VIA[Delta192] (Bellen)")
names(flies.labs) <- c("iPLA2-VIA Control", "iPLA2-VIA[Delta174]", "iPLA2-VIA[Delta192]")

bellen_mutant_192_taxon <- bellen_mutant_192_taxon + facet_grid(.~get(var1), drop = TRUE, space = "free", scales = "free", labeller = labeller(.cols = flies.labs)) +
  theme(
      strip.text.x = element_text(
        size = 30, color = "black", face = "bold"
        )
      )+ 
  xlab("Fly Samples") + 
  ylab("Fractional Abundance") +
  labs(fill = "Order Level Taxon")

plot(bellen_mutant_192_taxon)

```

### Bloomington Δ192 Taxon Chart

```{r Bloomington_192, exercise=TRUE}

file_path <- "Bloom_192"
urlpath <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/Bloom_192/"
mapper_file <- url("https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/Bloom_192/call_mapper.tsv")

var1 = "genotype"
taxonomic_level = "genus"
rpa_in_chart = 0.01
the_two_var = "onevar"
the_id = "X.SampleID"
plot_color = c("black","black","yellow", "brown","blue","red")
plot_order = c("Providencia", "Pasteurellales","Lactobacillaceae", "Lactobacillus", "Acetobacter")
x_val = "genotype"
read_depth = 1565
map_path = "-call19"
legend_position = "none"

bloom_192_taxon <- make_taxon_plots_1var(urlpath = urlpath, file_path = file_path,	map_path = map_path, mapper_file = mapper_file, var1 = var1,	taxonomic_level = taxonomic_level,	rpa_in_chart = rpa_in_chart,	the_two_var = "onevar",	the_id = "X.SampleID",	plot_color = plot_color,	plot_order = plot_order,	x_val = x_val,read_depth = read_depth, legend_position = legend_position)

flies.labs <- c("iPLA2-VIA", "iPLA2-VIA[Delta174]", "iPLA2-VIA[Delta192] (Bloomington)")
names(flies.labs) <- c("iPLA2-VIA Control", "iPLA2-VIA[Delta174]", "iPLA2-VIA[Delta192]")

bloom_192_taxon <- bloom_192_taxon + facet_grid(.~get(var1), drop = TRUE, space = "free", scales = "free", labeller = labeller(.cols = flies.labs)) +
  theme(
      strip.text.x = element_text(
        size = 30, color = "black", face = "bold"
        )
      )+ 
  xlab("Fly Samples") + 
  ylab("Fractional Abundance") +
  labs(fill = "Order Level Taxon")

plot(bloom_192_taxon)

```

### Δ192 PERMANOVA and PCoA Comparison

```{r 192_perm_pcoa, exercise=TRUE}

file_path <- "iPLA192_noControl"
urlpath <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/iPLA192_noControl/"
mapper_file <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/iPLA192_noControl/call_mapper.tsv"

map_file <- read.table(mapper_file,comment.char = "", header=T, fill=T, sep="\t") %>% mutate(X.SampleID = gsub(pattern = "-",replacement = "-",x = X.SampleID))

flies_weighted <- read.table(url(paste(urlpath,'core-metrics-results-',file_path,'/weighted_unifrac_distance_matrix/distance-matrix.tsv',sep="")), header=T, sep="\t")
flies_weighted_dm <- as.dist(flies_weighted[,2:dim(flies_weighted)[2]])
flies_unifrac_table <- flies_weighted %>% dplyr::select(X) %>% left_join(map_file, by=c("X"="X.SampleID")) %>% droplevels()

table <- adonis2(flies_weighted_dm ~ lab * vial_model, flies_unifrac_table, permutations=1000)
perm_192_noControl <- tableGrob(round(table, 2), theme=ttheme_minimal())
table <- tidy(round(table,2))
print("PERMANOVA analysis:")
print(table)

i192_lab_p <- table %>%
  filter(term == "lab") %>%
  pull(p.value) %>%
  round(3)
i192_lab_r <- table %>%
  filter(term == "lab") %>%
  pull(R2) %>%
  round(3)

var1 = "lab"
var1_colors <-c("blue","red")
legend_position = "bottom"
labs = c("Bellen", "Bloomington")
folder_name = paste0(file_path, "/weighted_unifrac")
title_name_add <- "weighted Unifrac"

PCoA_192_noControl <- pcoa_1var(urlpath = urlpath, folder_name=folder_name, mapper_file = mapper_file, var1=var1, var1_colors=var1_colors, title_name_add=title_name_add, legend_position = legend_position)
PCoA_192_noControl <- PCoA_192_noControl + scale_color_manual(name="Lab", labels=labs, values=var1_colors) + 
  scale_fill_manual(name="Lab", labels=labs, values=var1_colors) +
  ggtitle(paste0("R2=", i192_lab_r,", p=", i192_lab_p)) +
  theme(text = element_text(size = 30), axis.title = element_text(size = 30), plot.title = element_text(size = 30, hjust = 1)) +
  geom_point(size = 5) +
  stat_ellipse(aes(x=V2, y=V3,group= get(var1)), lwd = 1, level = .9, show.legend = F, type = "t", geom = "polygon", alpha = 0, inherit.aes=T)

plot(PCoA_192_noControl)
```

### Figure Creation

``` {r figure3, exercise=TRUE}

jpeg(h=800, w=1200, "Figure_3.jpg", units = "px", quality = 100, pointsize = 1)
	grid.arrange(
		bellen_mutant_192_taxon + labs(tag = "A") + theme(plot.tag = element_text(size = 45)), bloom_192_taxon + labs(tag = "B") + theme(plot.tag = element_text(size = 45)), PCoA_192_noControl + labs(tag = "C") + theme(plot.tag = element_text(size = 45)), perm_192_noControl,
		widths = c(1.5,1.5),	
   	heights = c(1, 1),
		layout_matrix=rbind(
		  c(1,2),
		  c(3,4))
		)
dev.off()

```

## Figure 4

*All sections of this figure were created separately and then brought together with grid arrange. This figure shows the overall and individual lab bacterial richness measures*

### Overall Richness Analysis

```{r all_richness, exercise=TRUE}

file_path <- "All"
urlpath <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/No_Wol/All/"
mapper_file <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/No_Wol/All/call_mapper.tsv"

map_file <- read.table(mapper_file,comment.char = "", header=T, fill=T, sep="\t") %>% mutate(X.SampleID = gsub(pattern = "-",replacement = "-",x = X.SampleID))

shannon_file <- read.table(url(paste(urlpath,'core-metrics-results-',file_path,'/shannon_richness/alpha-diversity.tsv',sep="")), header=T, sep="\t") %>% mutate(X=as.character(X))
shannon_table <- shannon_file %>% left_join(map_file, by=c("X"="X.SampleID"))

alpha <- shannon_table %>% group_by(genotype_class) %>% summarise(X, richness = mean(shannon_entropy)) %>% ungroup()

shannon_table <- shannon_table %>% left_join(alpha, "X" = "X")
shannon_table <- dplyr::select(shannon_table,X, genotype, genotype_class, richness, shannon_entropy)
wilcox <- wilcox.test(shannon_table$shannon_entropy~shannon_table$genotype_class)
shannon_table <- shannon_table %>% group_by(genotype_class) %>% summarise(X, genotype, genotype_class, richness, shannon_entropy, SEM = (sd(shannon_entropy) / sqrt(length(shannon_entropy)))) %>% ungroup()
shannon_table <- dplyr::select(shannon_table, genotype, genotype_class, richness, SEM) %>% unique()

ALL_shannon_plot <- ggplot(shannon_table, aes(x=genotype_class, y=richness)) + 
             geom_bar(position=position_dodge(), stat="identity") +
             geom_errorbar(aes(ymin=richness+SEM, ymax=richness-SEM),
                           width=.2,# Width of the error bars
                           position=position_dodge(.9)) +
             coord_cartesian(ylim=c((min(shannon_table$richness-shannon_table$SEM)*.5),max(shannon_table$richness+shannon_table$SEM)*1.15)) + 
             scale_x_discrete(breaks=c("control", "mutant"), labels=c("Control", "Mutant")) +
             theme_cowplot() +
             theme(axis.text=element_text(size=30),
                   axis.title = element_text(size = 30),
                   panel.background = element_blank(),
                   axis.line = element_line(), 
                   axis.ticks=element_line(),
                   title=element_text(size=20),
                   axis.title.x = element_blank(),
                   legend.position = "none") +
             labs(y="Richness",x=str_to_title("genotype"),title= paste0("p-value: ", toString(round(wilcox$p.value, 3))), tag = "A")

plot(ALL_shannon_plot)

```

### Bellen Richness Analysis

```{r bellen_richness, exercise=TRUE}

file_path <- "Bellen"
urlpath <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/No_Wol/Bellen/"
mapper_file <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/No_Wol/Bellen/call_mapper.tsv"

map_file <- read.table(mapper_file,comment.char = "", header=T, fill=T, sep="\t") %>% mutate(X.SampleID = gsub(pattern = "-",replacement = "-",x = X.SampleID))

shannon_file <- read.table(url(paste(urlpath,'core-metrics-results-',file_path,'/shannon_richness/alpha-diversity.tsv',sep="")), header=T, sep="\t") %>% mutate(X=as.character(X))
shannon_table <- shannon_file %>% left_join(map_file, by=c("X"="X.SampleID"))

alpha <- shannon_table %>% group_by(genotype_class) %>% summarise(X, richness = mean(shannon_entropy)) %>% ungroup()

shannon_table <- shannon_table %>% left_join(alpha, "X" = "X")
shannon_table <- dplyr::select(shannon_table,X, genotype, genotype_class, richness, shannon_entropy)
wilcox <- wilcox.test(shannon_table$shannon_entropy~shannon_table$genotype_class)
shannon_table <- shannon_table %>% group_by(genotype_class) %>% summarise(X, genotype, genotype_class, richness, shannon_entropy, SEM = (sd(shannon_entropy) / sqrt(length(shannon_entropy)))) %>% ungroup()
shannon_table <- dplyr::select(shannon_table, genotype, genotype_class, richness, SEM) %>% unique()

Bellen_shannon_plot <- ggplot(shannon_table, aes(x=genotype_class, y=richness)) + 
             geom_bar(position=position_dodge(), stat="identity", width = .5) +
             geom_errorbar(aes(ymin=richness+SEM, ymax=richness-SEM),
                           width=.2, # Width of the error bars
                           position=position_dodge(.9)) +
             coord_cartesian(ylim=c((min(shannon_table$richness-shannon_table$SEM)*.5),max(shannon_table$richness+shannon_table$SEM)*1.15)) +
             scale_x_discrete(breaks=c("control", "mutant"), labels=c("Control", "Mutant")) +
             theme_cowplot() +
             theme(axis.text=element_text(size=30),
                   axis.title = element_text(size = 30),
                   panel.background = element_blank(),
                   axis.line = element_line(), 
                   axis.ticks=element_line(),
                   title=element_text(size=20),
                   axis.title.x = element_blank(),
                   legend.position = "none") +
             labs(y="Richness",x=str_to_title("genotype"),title= paste0("p-value: ", toString(round(wilcox$p.value, 3))), tag = "B")

plot(Bellen_shannon_plot)
```

### Pallanck Richness Analysis

```{r pallanck_richness, exercise=TRUE}

file_path <- "Pallanck"
urlpath <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/No_Wol/Pallanck/"
mapper_file <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/No_Wol/Pallanck/call_mapper.tsv"

map_file <- read.table(mapper_file,comment.char = "", header=T, fill=T, sep="\t") %>% mutate(X.SampleID = gsub(pattern = "-",replacement = "-",x = X.SampleID))

shannon_file <- read.table(url(paste(urlpath,'core-metrics-results-',file_path,'/shannon_richness/alpha-diversity.tsv',sep="")), header=T, sep="\t") %>% mutate(X=as.character(X))
shannon_table <- shannon_file %>% left_join(map_file, by=c("X"="X.SampleID"))

alpha <- shannon_table %>% group_by(genotype) %>% summarise(X, richness = mean(shannon_entropy)) %>% ungroup()

shannon_table <- shannon_table %>% left_join(alpha, "X" = "X")
shannon_table <- dplyr::select(shannon_table,X, genotype, genotype_class, richness, shannon_entropy)
wilcox <- wilcox.test(shannon_table$shannon_entropy~shannon_table$genotype_class)
shannon_table <- shannon_table %>% group_by(genotype) %>% summarise(X, genotype, genotype_class, richness, shannon_entropy, SEM = (sd(shannon_entropy) / sqrt(length(shannon_entropy)))) %>% ungroup()
shannon_table <- dplyr::select(shannon_table, genotype, genotype_class, richness, SEM) %>% unique()



Pallanck_shannon_plot <- ggplot(shannon_table, aes(x=factor(genotype, levels = c("parkrvA (Control)", "park25")), y=richness)) + 
             geom_bar(position=position_dodge(), stat="identity", width = .5) +
             geom_errorbar(aes(ymin=richness+SEM, ymax=richness-SEM),
                           width=.2, # Width of the error bars
                           position=position_dodge(.9)) +
             coord_cartesian(ylim=c((min(shannon_table$richness-shannon_table$SEM)*.5),max(shannon_table$richness+shannon_table$SEM)*1.15)) +
             scale_x_discrete(breaks=c("parkrvA (Control)", "park25"), labels=c("Control", "Mutant")) +
             theme_cowplot() +
             theme(axis.text=element_text(size=30),
                   axis.title = element_text(size = 30),
                   panel.background = element_blank(),
                   axis.line = element_line(), 
                   axis.ticks=element_line(),
                   title=element_text(size=20),
                   axis.title.x = element_blank(),
                   legend.position = "none") +
             labs(y="Richness",x=str_to_title("genotype"),title= paste0("p-value: ", toString(round(wilcox$p.value, 3))), tag = "C")

plot(Pallanck_shannon_plot)
```

### Stanford Richness Analysis

```{r stanford_richness, exercise=TRUE}

file_path <- "Stanford"
urlpath <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/No_Wol/Stanford/"
mapper_file <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/No_Wol/Stanford/call_mapper.tsv"

map_file <- read.table(mapper_file,comment.char = "", header=T, fill=T, sep="\t") %>% mutate(X.SampleID = gsub(pattern = "-",replacement = "-",x = X.SampleID))

shannon_file <- read.table(url(paste(urlpath,'core-metrics-results-',file_path,'/shannon_richness/alpha-diversity.tsv',sep="")), header=T, sep="\t") %>% mutate(X=as.character(X))
shannon_table <- shannon_file %>% left_join(map_file, by=c("X"="X.SampleID"))

alpha <- shannon_table %>% group_by(genotype) %>% summarise(X, richness = mean(shannon_entropy)) %>% ungroup()

shannon_table <- shannon_table %>% left_join(alpha, "X" = "X")
shannon_table <- dplyr::select(shannon_table,X, genotype, genotype_class, richness, shannon_entropy)
wilcox <- wilcox.test(shannon_table$shannon_entropy~shannon_table$genotype_class)
shannon_table <- shannon_table %>% group_by(genotype) %>% summarise(X, genotype, genotype_class, richness, shannon_entropy, SEM = (sd(shannon_entropy) / sqrt(length(shannon_entropy)))) %>% ungroup()
shannon_table <- dplyr::select(shannon_table, genotype, genotype_class, richness, SEM) %>% unique()



Stanford_shannon_plot <- ggplot(shannon_table, aes(x=factor(genotype, levels = c("pink1 RV FM6 (Control)", "pink1 B9 FM6 Mutants")), y=richness)) + 
             geom_bar(position=position_dodge(), stat="identity", width = .5) +
             geom_errorbar(aes(ymin=richness+SEM, ymax=richness-SEM),
                           width=.2,# Width of the error bars
                           position=position_dodge(.9)) +
             coord_cartesian(ylim=c((min(shannon_table$richness-shannon_table$SEM)*.5),max(shannon_table$richness+shannon_table$SEM)*1.15)) + 
             scale_x_discrete(breaks=c("pink1 RV FM6 (Control)", "pink1 B9 FM6 Mutants"), labels=c("Control", "Mutant")) +
             theme_cowplot() +
             theme(axis.text=element_text(size=30),
                   axis.title = element_text(size = 30),
                   panel.background = element_blank(),
                   axis.line = element_line(), 
                   axis.ticks=element_line(),
                   title=element_text(size=20),
                   axis.title.x = element_blank(),
                   legend.position = "none") +
             labs(y="Richness",x=str_to_title("genotype"),title= paste0("p-value: ", toString(round(wilcox$p.value, 3))), tag = "D")

plot(Stanford_shannon_plot)
```

### UCLA Richness Analysis

```{r ucla_richness, exercise=TRUE}

file_path <- "UCLA"
urlpath <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/No_Wol/UCLA/"
mapper_file <- "https://raw.githubusercontent.com/emarsh25/parkFliesMicrobiome/master/Files/Analyses/No_Wol/UCLA/call_mapper.tsv"

map_file <- read.table(mapper_file,comment.char = "", header=T, fill=T, sep="\t") %>% mutate(X.SampleID = gsub(pattern = "-",replacement = "-",x = X.SampleID))

shannon_file <- read.table(url(paste(urlpath,'core-metrics-results-',file_path,'/shannon_richness/alpha-diversity.tsv',sep="")), header=T, sep="\t") %>% mutate(X=as.character(X))
shannon_table <- shannon_file %>% left_join(map_file, by=c("X"="X.SampleID"))

alpha <- shannon_table %>% group_by(genotype) %>% summarise(X, richness = mean(shannon_entropy)) %>% ungroup()

shannon_table <- shannon_table %>% left_join(alpha, "X" = "X")
shannon_table <- dplyr::select(shannon_table,X, genotype, genotype_class, richness, shannon_entropy)
wilcox <- wilcox.test(shannon_table$shannon_entropy~shannon_table$genotype_class)
shannon_table <- shannon_table %>% group_by(genotype) %>% summarise(X, genotype, genotype_class, richness, shannon_entropy, SEM = (sd(shannon_entropy) / sqrt(length(shannon_entropy)))) %>% ungroup()
shannon_table <- dplyr::select(shannon_table, genotype, genotype_class, richness, SEM) %>% unique()

UCLA_shannon_plot <- ggplot(shannon_table, aes(x=genotype, y=richness)) + 
             geom_bar(position=position_dodge(), stat="identity", width = .5) +
             geom_errorbar(aes(ymin=richness+SEM, ymax=richness-SEM),
                           width=.2,# Width of the error bars
                           position=position_dodge(.9)) +
             coord_cartesian(ylim=c((min(shannon_table$richness-shannon_table$SEM)*.5),max(shannon_table$richness+shannon_table$SEM)*1.15)) + 
             scale_x_discrete(breaks=c("PG (Control)", "pink15"), labels=c("Control", "Mutant")) +
             theme_cowplot() +
             theme(axis.text=element_text(size=30),
                   axis.title = element_text(size=30),
                   panel.background = element_blank(),
                   axis.line = element_line(), 
                   axis.ticks=element_line(),
                   title=element_text(size=20),
                   axis.title.x = element_blank(),
                   legend.position = "none") +
             labs(y="Richness",x=str_to_title("genotype"),title= paste0("p-value: ", toString(round(wilcox$p.value, 3))), tag = "E")

plot(UCLA_shannon_plot)
```

### Figure Creation

```{r figure4, exercise=TRUE}

jpeg(h=1200, w=1600, "Figure_4.jpg", units = "px", quality = 100, pointsize = 1)
	grid.arrange(
		UCLA_shannon_plot + labs(tag = "E") + theme(plot.tag = element_text(size = 45), title = element_text(size = 30)),Stanford_shannon_plot + labs(tag = "D") + theme(plot.tag = element_text(size = 45), title = element_text(size = 30)),Bellen_shannon_plot + labs(tag = "B") + theme(plot.tag = element_text(size = 45), title = element_text(size = 30)),Pallanck_shannon_plot + labs(tag = "C") + theme(plot.tag = element_text(size = 45), title = element_text(size = 30)), ALL_shannon_plot + labs(tag = "A") + theme(plot.tag = element_text(size = 45), title = element_text(size = 30)),
		widths = c(1,1,1),	
   	heights = c(1,1),
		layout_matrix=rbind(
		  c(5,3, 4),
			c(5,2,1))
		)
dev.off()
```
